README.md

```diff
@@ -1 +1,343 @@
+
+Your actions need not adhere to these requirements, but only those that do adhere can be considered
+for inclusion in `chef-actions`.
+
+Any unhandled exception in an action is re-raised, but only after invoking `notify(:error, exception)`.
+This gives the caller a chance to perform any cleanup in the notification handler - particularly helpful
+when the action is running on a different thread, in which case the exception may never be seen by
+your main thread (depending on your Ruby configuratoin).
+
+To define an action, create a class that inherits from `ChefCore::Actions::Base` and implement
+`perform_action`. Here's an example:
+
+### Simple Example
+```ruby
+require 'chef-core/actions'
+module MyApp
+  module Actions
+    class FindSomething < ChefCore::Actions::Base
+
+      def perform_action
+        notify(:looking_for_something, config[:search_criteria])
+
+        # Just sleep instead of doing anything. For future peformance
+        # improvement, reduce the sleep time.
+        sleep(1)
+
+        notify(:found, config[:search_criteria], "/home")
+
+        raise "Ooops"
+      end
+    end
+  end
+end
+```
+
+To invoke the action:
+
+```ruby
+require 'my_app/actions/find_something'
+
+criteria = gets("What should I seach for?").chomp
+action = LookForSomething.new(search_criteria: criteria)
+action.run do |event, *args|
+  case event
+  when :looking
+    puts "Searching for #{args[0]}"
+  when :found
+    puts "I found it! You can pick it up here: #{args[0]}
+  when :error
+    # Make sure we define :error - the framework notifies with this
+    # an perform_action raises an unhandled exception
+    puts "I'm sorry, something happened: #{args[0].to_s}
+end
+
+```
+
+Output:
+
+```
+What should I search for?
+blah
+
+Searching for blah
+I found it! You can pick it up here: /home
+I'm sorry, something happened: Ooops
+```
+
+
+### Real-life Usage
+
+[chef-apply](https://github.com/chef/chef-apply/blob/2a2b5d75641bc7fd0aaf3d12dd195a57cbdb9180/lib/chef_apply/cli.rb#L148) makes use
+of actions in a multi-threaded CLI that can perform actions simultaneously across multiple hosts.
+
+
+## CLI User Experience
+
+More information coming here as we determine if what we have now is the right shape for future CLI tool
+development.
+
+
+## Contributing/Development
+
+Please read our [Community Contributions Guidelines](https://docs.chef.io/community_contributions.html), and ensure you are signing all your commits with DCO sign-off.
+
+The general development process is:
+
+1. Fork this repo and clone it to your workstation.
+2. Create a feature branch for your change.
+3. Write code and tests.
+4. Push your feature branch to GitHub and open a pull request against master.
+
+Once your repository is set up, you can start working on the code. We do utilize RSpec for test driven development, so you'll need to get a development environment running. Follow the above procedure ("Installing from Git") to get your local copy of the source running.
```

Comment:
Can you use an internal link for "Installing from Git"? I couldn't find the section that you were referring to by Header Title.

Reply:
Yep, I'll get this straightened out.